include/eve/arch/riscv/tags.hpp

```diff
@@ -0,0 +1,115 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/cardinals.hpp>
+#include <eve/arch/cpu/tags.hpp>
+#include <eve/arch/riscv/predef.hpp>
+#include <eve/concept/scalar.hpp>
+#include <eve/detail/meta.hpp>
+
+#include <bit>
+
+namespace eve
+{
+//================================================================================================
+// ABI tags for all RISCV bits SIMD registers
+//================================================================================================
+template<std::size_t Size> struct rvv_abi_
+{
+  static constexpr std::size_t MaxLmul         = 8;
+  static constexpr std::size_t bits            = Size * MaxLmul;
+  static constexpr std::size_t bytes           = bits / 8;
+  static constexpr bool        is_wide_logical = false;
+
+  template<typename Type>
+  static constexpr bool is_full = ((Type::size() * sizeof(typename Type::value_type)) >= bytes);
+
+  template<typename Type>
+  static constexpr std::size_t fundamental_cardinal = Size / 8 / sizeof(Type);
+
+  template<typename Type>
+  static constexpr std::size_t expected_cardinal = Size / 8 / sizeof(Type) * MaxLmul;
```

Comment:
there is some confusion here.

I thought fundamental cardinal should be Size / 8 / sizeof(Type) / MaxLmul to represent negative multipliers

Reply:
To make this, cardinals should be type-dependent. Here I described how types are formed and why some negative multipliers can not be used with some element types https://github.com/jfalcou/eve/pull/1920#discussion_r1720966727. For this case let's assume you want to use `std::uint64_t` as element type. If your vlen is 64, you can not use part of vector register to represent even one 64 bit element, so no fractional LMUL for this element type available. There I use smallest vector register type that could fit for every element type.

Reply:
Oh, it seems that they are already type-dependent. Fixed.